controllers/operatorpipeline_controller.go

```diff
@@ -64,6 +72,48 @@ func (r *OperatorPipelineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
+	// Check if the OperatorPipeline instance is marked to be deleted, which is
+	// indicated by the deletion timestamp being set.
+	isOperatorPipelineMarkedToBeDeleted := currentPipeline.GetDeletionTimestamp() != nil
+	if isOperatorPipelineMarkedToBeDeleted {
+		if controllerutil.ContainsFinalizer(currentPipeline, operatorPipelineFinalizer) {
+			namespacePipelines := &certv1alpha1.OperatorPipelineList{}
+			listOptions := client.InNamespace(currentPipeline.Namespace)
+			if err := r.Client.List(ctx, namespacePipelines, listOptions); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			// if the length is 1 we know this is the last CR in a given namespace and can remove the ClusterRoleBinding
+			// associated with this namespace
+			if len(namespacePipelines.Items) == 1 {
+				if err := r.deleteClusterRoleBinding(ctx, log, currentPipeline.Namespace); err != nil {
+					return ctrl.Result{}, err
+				}
+			}
+
+			clusterPipelines := &certv1alpha1.OperatorPipelineList{}
+			if err := r.Client.List(ctx, clusterPipelines); err != nil {
```

Comment:
Just so I'm clear, because this doesn't have ListOptions, we're querying for OperatorPipelineList and the items will span all namespaces?

Reply:
Yes correct, I wasn't sure if querying twice was the best option, but since the `ClusterRoleBinding` is tied to a namespace `ServiceAccount` and references the `namespace` I ended up with this approach, though I'm really open to other ideas if there are any.

Maybe this needs comments on the `List` calls, right now I have the explanation on the `if` logic after these calls. Thoughts?

Reply:
Maybe just a small blurb explaining the intent here (e.g. move the "if this is instance is the last instance cluster-wide, also delete <N>" here)

Reply:
I added two more comments, please take a look at all the comments and let me know if they are clear, or if having this many is confusing.